databricks/koalas/namespace.py

```diff
@@ -1488,17 +1497,40 @@ def concat(objs, axis=0, join='outer', ignore_index=False):
 
     # Series, Series ...
     # We should return Series if objects are all Series.
-    should_return_series = all(map(lambda obj: isinstance(obj, Series), objs))
+    if axis == 0:
+        should_return_series = all(map(lambda obj: isinstance(obj, Series), objs))
 
     # DataFrame, Series ... & Series, Series ...
     # In this case, we should return DataFrame.
     new_objs = []
-    for obj in objs:
+    for idx, obj in enumerate(objs):
         if isinstance(obj, Series):
-            obj = obj.rename('0').to_dataframe()
+            if axis == 0:
+                obj = obj.rename('0').to_dataframe()
+            else:
+                if obj.name == '0':
+                    obj = obj.rename(str(idx)).to_dataframe()
+                else:
+                    obj = obj.to_dataframe()
         new_objs.append(obj)
     objs = new_objs
 
+    if axis == 1:
```

Comment:
For `ignore_index`, I think we can explicitly throw an exception for now when axis is 1.